test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +92,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (command.isMutating()) {
```

Comment:
This might not be much useful as we thought, cause it could not verify that only mutating command can save file. I suggest adding an else here to handle the other commands, in which checking them could not modify the file (e.g. check the updated time)